docs/data_classes.rst

```diff
@@ -0,0 +1,301 @@
+
+  >>> from ndcube import NDCube
+  >>> my_cube = NDCube(data, input_wcs)
+
+The data array is stored in ``mycube.data`` while the WCS object is stored in ``my_cube.wcs``.  However, when manipulating/slicing the data it is better to slice the object as a whole.  (See section on :ref:`ndcube_slicing`.)  So the ``.data`` attribute should only be used to access specific raw data values.
+
+Thanks to its inheritance from `astropy.nddata.NDData`, `~ndcube.NDCube` can also hold additional supplementary data including: metadata located at `NDCube.meta`;
+an uncertainty array located at `NDCube.uncertainty` (subclass of `astropy.nddata.NDUncertainty`) describing the uncertainty of each data array value;
+a data unit (`astropy.units.Unit` or unit `str`);
+and a mask (boolean array), located at `NDCube.mask`, marking reliable and unreliable pixels.
+Note that in keeping the convention of `numpy.ma.masked_array` ``True`` means that the corresponding data array axis is masked, i.e. it is bad data, while ``False`` signifies good data.
+
+.. code-block:: python
+
+  >>> import astropy.units as u
+  >>> from astropy.nddata import StdDevUncertainty
+  >>> uncertainty = StdDevUncertainty(np.sqrt(np.abs(data)))
+  >>> mask = np.zeros_like(my_cube.data, dtype=bool)
+  >>> meta = {"Description": "This is example NDCube metadata."}
+  >>> my_cube = NDCube(data, input_wcs, uncertainty=uncertainty, mask=mask,
+  ...                  meta=meta, unit=u.ct)
+
+Dimensions and Physical Types
+-----------------------------
+
+`~ndcube.NDCube` has useful properties for inspecting its data shape and
+axis types, `~ndcube.NDCube.dimensions` and `~ndcube.NDCube.array_axis_physical_types`.
+
+.. code-block:: python
+
+  >>> my_cube.dimensions
+  <Quantity [3., 4., 5.] pix>
+  >>> my_cube.array_axis_physical_types
+  [('custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon'),
+   ('custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon'),
+   ('em.wl',)]
+
+`~ndcube.NDCube.dimensions` returns an `~astropy.units.Quantity` of pixel units giving the length of each dimension in the `~ndcube.NDCube` while `~ndcube.NDCube.array_axis_physical_types` returns tuples of strings denoting the types of physical properties represented by each array axis.  The tuples are arranged in array axis order.  As more than one physical type can be associated with an array axis, the length of each tuple can be greater than 1.  This is the case for the 1st and 2nd array array axes which are associated with the coupled world axes of helioprojective latitude and longitude. The axis names are in accordance with the International Virtual Observatory Alliance (IVOA)
+`UCD1+ controlled vocabulary <http://www.ivoa.net/documents/REC/UCD/UCDlist-20070402.html>`_.
+
+`~ndcube.NDCube` provides many helpful features, specifically regarding coordinate transformations, slicing and visualization.  See the :ref:`cube_coordinates`, :ref:`cube_slicing` and :ref:`cube_plotting` sections.
```

Comment:
similarly here the `cube_slicing` and `cube_plotting` not linking correctly on readthedocs